numpyro/primitives.py

```diff
@@ -101,12 +125,20 @@ def sample(name, fn, obs=None, rng_key=None, sample_shape=(), infer=None):
         for inference algorithms. For example, if `fn` is a discrete distribution,
         setting `infer={'enumerate': 'parallel'}` to tell MCMC marginalize
         this discrete latent site.
+    :param numpy.ndarray obs_mask: Optional boolean array mask of shape
+        broadcastable with ``fn.batch_shape``. If provided, events with
+        mask=True will be conditioned on ``obs`` and remaining events will be
+        imputed by sampling. This introduces a latent sample site named ``name
+        + "_unobserved"`` which should be used by guides.
     :return: sample from the stochastic `fn`.
     """
     # if there are no active Messengers, we just draw a sample and return it as expected:
     if not _PYRO_STACK:
         return fn(rng_key=rng_key, sample_shape=sample_shape)
 
+    if obs_mask is not None:
+        return _masked_observe(name, fn, obs, obs_mask, rng_key=None, sample_shape=(), infer=None)
```

Comment:
Shouldn't this pass through arguments? I'm surprised the `rng_key` bit didn't error 😕 
```py
return _masked_observe(name, fn, obs, obs_mask, rng_key=rng_key, sample_shape=sample_shape, infer=infer)
```

Reply:
Sorry, I missed this comment. I'll address this too.